<launch>
    <!-- The length of each bag file in seconds.
         Acceptable values: positive integers
         If not given defaults to 30 seconds
         If an invalid value is given the node will exit and log an error message -->
    <arg name="bag_rollover_time" default="5" />

    <!-- The length of time recordings are kept. Any rosbag older than this time will be deleted.
         Acceptable values: positive integers
         If not given defaults to 300 seconds
         If an invalid value is given the node will exit and log an error message -->
    <arg name="max_record_time" default="300" />


    <!-- Directory where rosbags will be written.
         Defaults to ~/.ros/rosbag_uploader/ -->
    <arg name="write_directory" default="$(env HOME)/.ros/rosbag_uploader/" />

    <!-- Time in seconds for server connection time out and
         s3 upload action time out -->
    <arg name="upload_timeout" default="3600" />

    <!-- The topics that should be recorded into the rosbag.
         Acceptable values: any list of strings or an empty list.
         e.g. '/topic1 /topic2 /topic3'
         If list is empty or no value is given defaults to all topics.
         If a topic in the list doesn't exist when recording starts
         no warning will be given. Behavior here depends on ROSBag implementaiton. -->
    <arg name="topics_to_record" default='-a' />

    <arg name="s3_bucket"/>

    <include file="$(find rosbag_cloud_recorders)/launch/rolling_recorder.launch">
      <arg name="bag_rollover_time" value="$(arg bag_rollover_time)"/>
      <arg name="upload_timeout" value="$(arg upload_timeout)"/>
      <arg name="write_directory" value="$(arg write_directory)"/>
      <arg name="max_record_time" value="$(arg max_record_time)"/>
      <arg name="topics_to_record" value="$(arg topics_to_record)"/>
    </include>

    <include file="$(find s3_file_uploader)/launch/s3_file_uploader.launch">
      <arg name="s3_bucket" value="$(arg s3_bucket)"/>
    </include>

</launch>